packages/ui/aurora/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,200 @@
+  const popperScope = usePopperScope(__scopeComboBox);
+
+  // https://www.downshift-js.com/use-combobox
+  const comboProps = useCombobox<T>({
+    items: items ?? [],
+    selectedItem: value ?? null,
+    itemToString: (selectedItem) => (selectedItem ? adapter(selectedItem).text : ''),
+    onInputValueChange: ({ inputValue }) => onInputChange?.(inputValue),
+    onSelectedItemChange: ({ selectedItem }) => onChange?.(selectedItem === null ? undefined : selectedItem),
+  });
+
+  return (
+    <PopperPrimitive.Root {...popperScope}>
+      <ComboBoxProvider scope={__scopeComboBox} adapter={adapter} items={items ?? []} {...comboProps}>
+        <div className={tx('combobox.root', 'combobox__root', {}, classNames)}>
+          <ComboBoxAnchor placeholder={placeholder} />
+          <ComboBoxContent />
+        </div>
+      </ComboBoxProvider>
+    </PopperPrimitive.Root>
+  );
+};
+
+//
+// Anchor
+//
+
+const ANCHOR_NAME = 'ComboBoxAnchor';
+
+type ComboBoxAnchorProps = ThemedClassName<
+  ScopedProps<
+    PropsWithChildren<{
+      placeholder?: string;
+    }>
+  >
+>;
+
+const ComboBoxAnchor = forwardRef<HTMLDivElement, ComboBoxAnchorProps>(
+  ({ __scopeComboBox, classNames, placeholder }: ComboBoxAnchorProps, forwardedRef) => {
+    const { tx } = useThemeContext();
+    const popperScope = usePopperScope(__scopeComboBox);
+    const { getInputProps, getToggleButtonProps, isOpen } = useComboBoxContext(ANCHOR_NAME, __scopeComboBox);
+
+    return (
+      <PopperPrimitive.Anchor asChild {...popperScope} ref={forwardedRef}>
+        <div className='flex items-center gap-1'>
+          <Input.Root>
+            <Input.TextInput
+              {...getInputProps()}
+              placeholder={placeholder}
+              variant='subdued'
+              classNames={tx('combobox.input', 'combobox__input', {}, classNames)}
+            />
+          </Input.Root>
+          <Button
+            {...getToggleButtonProps()}
+            aria-label='toggle menu'
+            variant='ghost'
+            classNames={tx('combobox.button', 'combobox__button', {}, classNames)}
+          >
+            {/* TODO(burdon): SelectPrimitive.Icon? */}
+            {(isOpen && <CaretUp />) || <CaretDown />}
+          </Button>
+        </div>
+      </PopperPrimitive.Anchor>
+    );
+  },
+);
+
+//
+// Content
+//
+
+const CONTENT_NAME = 'ComboBoxContent';
+
+type ComboBoxContentProps = ThemedClassName<ScopedProps<PropsWithChildren<{}>>>;
+
+const ComboBoxContent = forwardRef<HTMLDivElement, ComboBoxContentProps>(
+  ({ __scopeComboBox, classNames }: ComboBoxContentProps, forwardedRef) => {
+    const { tx } = useThemeContext();
+    const popperScope = usePopperScope(__scopeComboBox);
+    const { adapter, items, getMenuProps, getItemProps, highlightedIndex, selectedItem, isOpen } = useComboBoxContext(
+      CONTENT_NAME,
+      __scopeComboBox,
+    );
+
+    if (!isOpen) {
+      return null;
+    }
+
+    return (
+      <div>
```

Comment:
What purpose does this extra `div` serve? Can it be eliminated? If not, it needs to support `asChild` and needs a role.

Reply:
scrolling didn't work without it.